crates/driver/src/domain/competition/settled/mod.rs

```diff
@@ -0,0 +1,268 @@
+use {
+    super::{
+        auction,
+        order::{self, Side},
+    },
+    crate::{domain::eth, util::conv::u256::U256Ext},
+    number::conversions::big_rational_to_u256,
+    std::collections::HashMap,
+};
+
+/// Settlement built from onchain calldata.
+#[derive(Debug, Clone)]
+pub struct Settlement {
+    trades: Vec<Trade>,
+}
+
+impl Settlement {
+    pub fn new(trades: Vec<Trade>) -> Self {
+        Self { trades }
+    }
+
+    /// Score of a settlement as per CIP38
+    ///
+    /// Denominated in NATIVE token
+    pub fn score(
+        &self,
+        prices: &HashMap<eth::TokenAddress, auction::NormalizedPrice>,
+    ) -> Option<eth::TokenAmount> {
+        self.trades
+            .iter()
+            .map(|t| t.score(prices))
+            .try_fold(eth::TokenAmount(eth::U256::zero()), |acc, score| {
+                score.map(|score| acc + score)
+            })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Trade {
+    sell: eth::Asset,
+    buy: eth::Asset,
+    side: Side,
+    executed: order::TargetAmount,
+    prices: Prices,
+    policies: Vec<order::FeePolicy>,
+}
+
+impl Trade {
+    pub fn new(
+        sell: eth::Asset,
+        buy: eth::Asset,
+        side: Side,
+        executed: order::TargetAmount,
```

Comment:
Just to double check notation:
- `sell` and `buy` represent limit sell and buy amounts and we already assume that all signed fees are zero at this point.
- `executed` for a sell order is the amount a user sends to the protocol and for a buy order the amount a user receives from the protocol
- `executed` converted using `prices.custom` gives the amount a user receives for sell orders and the amount they send for buy orders.
Is that correct?